DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/EncryptProcessor.cs

```diff
@@ -0,0 +1,99 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dicom;
+using Dicom.IO.Buffer;
+using EnsureThat;
+using Microsoft.Health.DeID.SharedLib.Settings;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.DeID.SharedLib;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    /// <summary>
+    /// We use AES-CBC algorithm to transform the value with an encryption key, and then replace the original value with a Base64 encoded representation of the encrypted value. 
+    /// </summary>
+    public class EncryptProcessor : IAnonymizerProcessor
+    {
+        private readonly EncryptFunction _encryptFunction;
+        private static readonly HashSet<string> _supportedVR = Enum.GetNames(typeof(EncryptSupportedVR)).ToHashSet(StringComparer.InvariantCultureIgnoreCase);
+
+        public EncryptProcessor(JObject settingObject)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            var settingFactory = new AnonymizerSettingsFactory();
+            var encryptionSetting = settingFactory.CreateAnonymizerSetting<EncryptSetting>(settingObject);
+            _encryptFunction = new EncryptFunction(encryptionSetting);
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            try
+            {
+                if (item is DicomStringElement)
+                {
+                    var encryptedValues = ((DicomStringElement)item).Get<string[]>().Where(x => !string.IsNullOrEmpty(x)).Select(x => EncryptToBase64String(x));
```

Comment:
>EncryptToBase64String [](http://example.com/codeflow?start=134&length=21)

this is a minor problem. I guess you can always transfer the encrypted data to a string format like the key representation.